Long pollling fallback (#25621)
dotnet/docs: docs/core/compatibility/aspnet-core/6.0/blazor-long-polling-fallback.md

---
title: "Breaking change: Blazor server: Disable LongPolling fallback transport"

> description: "Learn about the breaking change in ASP.NET Core 6.0 where LongPolling is not used as a fallback transport utilized when WebSockets aren't available."

Reviewer: ```suggestion
description: "Learn about the breaking change in ASP.NET Core 6.0 where LongPolling is not used as a fallback transport when WebSockets aren't available."
```

ms.date: 08/12/2021
no-loc: [ Blazor, WebSocket, LongPolling ]
---
# Blazor server: Disable long polling fallback transport

> *LongPolling* is a fallback transport utilized when WebSockets aren't available. The LongPolling fallback can lead to a degraded user experience, so it has been removed. Both the client and server now support only WebSockets by default.

Reviewer: ```suggestion
Prior to .NET 6, *LongPolling* was a fallback transport utilized when WebSockets weren't available. The LongPolling fallback can lead to a degraded user experience, so it has been removed. Both the client and server now support only WebSockets by default.
```

## Version introduced

ASP.NET Core 6.0 RC 1

## Old behavior

If WebSockets are unavailable for a circuit (for example, due to network issues or browser incompatibility), LongPolling is used instead.

## New behavior

The following table shows the error message you'll receive for each combination of client and server configurations.

| Client | Server | Message |
|---|---|---|
| WS (without browser WS support) | WebSockets | `Unable to connect, please ensure you are using an updated browser that supports WebSockets.` |
| WS (with WS connection being rejected) | WebSockets | `Unable to connect, please ensure WebSockets are available. A VPN or proxy may be blocking the connection.` |
| WS | LongPolling | `An unhandled error has occurred.` Console Error: `Unable to initiate a SignalR connection to the server. This might be because the server is not configured to support WebSockets. To troubleshoot this, visit https://aka.ms/blazor-server-websockets-error.` |
| LongPolling | WebSockets | `An unhandled error has occurred.` |

## Reason for change

> This change was mode to improve the overall end-user experience by enforcing WebSocket use.

Reviewer: ```suggestion
This change was made to improve the overall end-user experience by enforcing WebSocket use.
```

Author: Good catches!
